src/Features/CSharp/Portable/CodeRefactorings/ConvertConversionOperators/ConvertConversionOperatorsFromAsRefactoringProvider.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.ConvertConversionOperators;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.ConvertConversionOperators
+{
+    /// <summary>
+    /// Refactor:
+    ///     var o = 1 as object;
+    ///
+    /// Into:
+    ///     var o = (object)1;
+    /// </summary>
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.ConvertConversionOperators), Shared]
+    internal partial class CSharpConvertConversionOperatorsFromAsRefactoringProvider
+        : AbstractConvertConversionOperatorsRefactoringProvider<BinaryExpressionSyntax>
+    {
+        [ImportingConstructor]
+        [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
+        public CSharpConvertConversionOperatorsFromAsRefactoringProvider()
+        {
+        }
+
+        protected override Task<ImmutableArray<BinaryExpressionSyntax>> FilterFromExpressionCandidatesAsync(
+            ImmutableArray<BinaryExpressionSyntax> asExpressions,
+            Document document,
+            CancellationToken cancellationToken)
```

Comment:
this doesn't feel like it needs to be async, even for other cases.  if something does need a SemanticModel, it could be passed in.

Reply:
Asking for a semantic model is effectively free. :)
First, it's at most 1-2 tiny allocations.  Second, in all scenarios it practically already exists and you'll get the cached one.

That said, here i don't think you need it.  It seems like you could do all this work both syntactically, and in the base type entirely (i.e.by using ISyntaxKinds).

Reply:
`SyntaxFacts.GetPartsOfCastExpression` does not support `CTypeExpressionSyntax` (only `DirectCastExpressionSyntax`). Is it save to add it?

Reply:
i didn't notice that.  in general i would expect this to be working on DirectCast, and not really on CType.  This is a feature that converts between `(type)` and `as`.  I would expect the VB equiv to be converting between Direct and Try casts.

Reply:
I did the change in 0f7c894. Can you please have a short look at that commit? (tests for the new providers pass, but I have not checked the whole solution with the change to GetPartsOfCastExpression

Reply:
> I would expect the VB equiv to be converting between Direct and Try casts.

Shall I change it?

Reply:
That would be nice :)

Reply:
Done dc0f6ad